src/test/java/org/dice_research/opal/launuts/polygons/parser/NutsParserMethodsTest.java

```diff
@@ -1,44 +1,64 @@
 package org.dice_research.opal.launuts.polygons.parser;
 
+import java.io.FileReader;
+import java.io.IOException;
+
 import org.json.simple.JSONArray;
 import org.json.simple.JSONObject;
+import org.json.simple.parser.ParseException;
 import org.junit.Assert;
 import org.junit.Test;
 
 public class NutsParserMethodsTest {
 
 	@Test
-	public void getInnerRings() {
-		double[][] arrays = { { 50.13351, 8.81796 }, { 50.07466, 8.8237 }, { 50.06241, 8.71783 }, { 50.13673, 8.78065 },
-				{ 50.13351, 8.81796 } };
-		JSONObject coordinates = new JSONObject();
-		coordinates.put("coordinates", arrays);
-		JSONArray contentItems = new JSONArray();
-		contentItems.add(coordinates);
-		JSONArray inner_rings = NutsParser.getInnerRings(contentItems);
-		System.out.println(inner_rings);
-	}
+	public void getInnerRings() throws IOException, ParseException {
+		// Read JSON array to pass in the method
+		NutsParser nutParser = new NutsParser();
+		nutParser.geojsonReader = new FileReader("src/test/resources/nuts_synthetic_one/test-nuts-three.geojson");
+		JSONObject rootObject = (JSONObject) nutParser.jsonParser.parse(nutParser.geojsonReader);
+		JSONArray feature = (JSONArray) rootObject.get("features");
+		JSONObject firstObject = (JSONObject) feature.get(0);
+		JSONObject jsonGeometry = (JSONObject) firstObject.get("geometry");
+		JSONArray coordinates = (JSONArray) jsonGeometry.get("coordinates");
 
-//	@Test
-//	public void hasThisNutsLeastNumberOfCoordinatesIfTrueThenAdd() {
-//	}
+		// Read JSON array to compare with the result
+		nutParser.geojsonReader = new FileReader("src/test/resources/nuts_synthetic_one/test-nuts-four.geojson");
+		JSONObject rootObject1 = (JSONObject) nutParser.jsonParser.parse(nutParser.geojsonReader);
+		JSONArray feature1 = (JSONArray) rootObject1.get("features");
+		JSONObject firstObject1 = (JSONObject) feature1.get(0);
+		JSONObject jsonGeometry1 = (JSONObject) firstObject1.get("geometry");
+		JSONArray coordinatesExpected = (JSONArray) jsonGeometry1.get("coordinates");
+		JSONArray inner_rings = nutParser.getInnerRings(coordinates);
```

Comment:
Move following to line no 24 for better understanding.

JSONArray inner_rings = nutParser.getInnerRings(coordinates);